### ISSUE TITLE
Own Keys are "not trusted signing keys"

### ISSUE THREAD

**Issue by franco-bez** [k9mail/k-9#2054]
### Expected behavior
When I open a Mail in the "Sent Mail" Folder that was encrypted and signed by my own GPG Key it should be OK and no Warning displayed.

### Actual behavior
Opening an E-Mail that was signed and encrypted by my own GPG Key (with K-9 and OpenKeychain) from the "Sent E-Mail Folder, I get the message "Safety Warning: This message was signed by an unsafe Key - Show message anyways"

### Steps to reproduce
1. Send an encrypted E-Mail
2. go the the "Sent E-Mails Folder and open the Mail you just sent
3. Read the Error Message

### Environment
K-9 Mail version: 5.202

Android version:4.4.2

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

---

First of all I want to tell you that the new K-9 Version made an enormous improvement in handling encrypted messages.
Great work :-) 

The Bug is merely a little annoying - why is my own Key "unsafe" ?
Why  does K-9 not trust my own identity ?

![screenshot_2017-01-14-10-08-23](https://cloud.githubusercontent.com/assets/4195055/21953850/1dc79bfc-da42-11e6-867e-85fbba832b13.png)

**Comment by philipwhiuk** [comment]
This is an Open Keychain decision. But it's probably worth documenting and understanding from a K-9 perspective.

**Comment by Valodim** [comment]
This is correct behavior, unfortunately our error reporting isn't very good. Your key is considered unsafe by OpenKeychain because of some of its parameters, e.g. if it's a 1024 bit key.

It's noteworthy that other implementations often don't care about these parameters and will happily tell you that a signature made with algorithms that are no longer considered secure is just fine. Since we are more fail fast in this regard, this puts us in the role of the messenger of bad news.

**Comment by franco-bez** [comment]
Hi Valodim, hi philipwhiuk,

thanks for the feedback.
Yes my key is rather an old one. Mabe it would be a good idea to renew it.
Strange that in Openkeychain itself I see no warning at all.
![screenshot_2017-01-14-14-20-13](https://cloud.githubusercontent.com/assets/4195055/21955184/d87e4ab8-da65-11e6-8585-e5a21d434fc4.png)
![screenshot_2017-01-14-14-20-40](https://cloud.githubusercontent.com/assets/4195055/21955185/de2411c8-da65-11e6-88e5-aeb06f4991b7.png)
![screenshot_2017-01-14-14-21-00](https://cloud.githubusercontent.com/assets/4195055/21955186/e2b82652-da65-11e6-860d-3624aa51ca6d.png)

**Comment by Valodim** [comment]
That's correct. The error reporting isn't very good, that unfortunately applies to OpenKeychain itself as well :)

**Comment by rugk** [comment]
Yes 1024bit DSA is insecure. You might report the missing warning in OpenKeychain to OpenKeyChain, so they can add one there.
